<?xml version="1.0" encoding="UTF-8"?>
<robot generator="Robot 5.0.1 (Python 3.10.2 on win32)" generated="20220908 18:07:29.944" rpa="false" schemaversion="3">
<suite id="s1" name="perfecto Test" source="C:\Users\vuchander\POD5_Mobile_libs\Tests\perfecto_Test.robot">
<test id="s1-t1" name="To check whether we can successfully login with correct Credentials" line="23">
<kw name="Open app">
<kw name="Open Application" library="AppiumLibrary">
<arg>${perfecturl}</arg>
<arg>${app_id}</arg>
<arg>securityToken=${token}</arg>
<arg>deviceName=${deviceid_android}</arg>
<arg>noReset=${True}</arg>
<arg>appPackage="chat.rocket.android"</arg>
<arg>browserName=mobileOS</arg>
<doc>Opens a new application to given Appium server.
Capabilities of appium server, Android and iOS,
Please check https://github.com/appium/appium/blob/master/docs/en/writing-running-appium/server-args.md
| *Option*            | *Man.* | *Description*     |
| remote_url          | Yes    | Appium server url |
| alias               | no     | alias             |</doc>
<kw name="Capture Page Screenshot" library="AppiumLibrary">
<doc>Takes a screenshot of the current page and embeds it into the log.</doc>
<msg timestamp="20220908 18:07:50.806" level="FAIL">No application is open</msg>
<status status="FAIL" starttime="20220908 18:07:50.806" endtime="20220908 18:07:50.821"/>
</kw>
<msg timestamp="20220908 18:07:50.821" level="WARN">Keyword 'Capture Page Screenshot' could not be run on failure: No application is open</msg>
<msg timestamp="20220908 18:07:50.821" level="FAIL">WebDriverException: Message: Failed to execute command appium.mode start: appiumModeStart command failed. Reason: handset server: application is not installed
Stacktrace:
    at org.openqa.selenium.remote.server.PerfectoMobileSession$Factory.convertToRuntimeException (PerfectoMobileSession.java:166)
    at org.openqa.selenium.remote.server.PerfectoMobileSession$Factory.apply (PerfectoMobileSession.java:150)
    at org.openqa.selenium.remote.server.PerfectoMobileActiveSessionFactory.lambda$apply$3 (PerfectoMobileActiveSessionFactory.java:117)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:193)
    at java.util.stream.ReferencePipeline$11$1.accept (ReferencePipeline.java:373)
    at java.util.stream.ReferencePipeline$2$1.accept (ReferencePipeline.java:175)
    at java.util.Spliterators$ArraySpliterator.tryAdvance (Spliterators.java:958)
    at java.util.stream.ReferencePipeline.forEachWithCancel (ReferencePipeline.java:126)
    at java.util.stream.AbstractPipeline.copyIntoWithCancel (AbstractPipeline.java:498)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:485)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:471)
    at java.util.stream.FindOps$FindOp.evaluateSequential (FindOps.java:152)
    at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.findFirst (ReferencePipeline.java:464)
    at org.openqa.selenium.remote.server.PerfectoMobileActiveSessionFactory.apply (PerfectoMobileActiveSessionFactory.java:120)
    at org.openqa.selenium.remote.server.NewSessionPipeline.lambda$null$2 (NewSessionPipeline.java:66)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:193)
    at java.util.stream.ReferencePipeline$2$1.accept (ReferencePipeline.java:175)
    at java.util.Collections$2.tryAdvance (Collections.java:4717)
    at java.util.stream.ReferencePipeline.forEachWithCancel (ReferencePipeline.java:126)
    at java.util.stream.AbstractPipeline.copyIntoWithCancel (AbstractPipeline.java:498)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:485)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:471)
    at java.util.stream.FindOps$FindOp.evaluateSequential (FindOps.java:152)
    at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.findFirst (ReferencePipeline.java:464)
    at org.openqa.selenium.remote.server.NewSessionPipeline.lambda$createNewSession$3 (NewSessionPipeline.java:69)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:193)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:193)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:193)
    at java.util.stream.DistinctOps$1$2.accept (DistinctOps.java:175)
    at java.util.stream.ReferencePipeline$2$1.accept (ReferencePipeline.java:175)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:193)
    at java.util.stream.ReferencePipeline$2$1.accept (ReferencePipeline.java:175)
    at java.util.stream.Streams$StreamBuilderImpl.tryAdvance (Streams.java:405)
    at java.util.stream.Streams$ConcatSpliterator.tryAdvance (Streams.java:728)
    at java.util.stream.ReferencePipeline.forEachWithCancel (ReferencePipeline.java:126)
    at java.util.stream.AbstractPipeline.copyIntoWithCancel (AbstractPipeline.java:498)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:485)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:471)
    at java.util.stream.FindOps$FindOp.evaluateSequential (FindOps.java:152)
    at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.findFirst (ReferencePipeline.java:464)
    at org.openqa.selenium.remote.server.NewSessionPipeline.createNewSession (NewSessionPipeline.java:72)
    at org.openqa.selenium.remote.server.commandhandler.BeginSession.execute (BeginSession.java:65)
    at org.openqa.selenium.remote.server.MobileWebDriverServlet.lambda$handle$2 (MobileWebDriverServlet.java:299)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:511)
    at java.util.concurrent.FutureTask.run (FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:617)
    at java.lang.Thread.run (Thread.java:748)</msg>
<status status="FAIL" starttime="20220908 18:07:30.219" endtime="20220908 18:07:50.821"/>
</kw>
<kw name="Sleep" library="BuiltIn">
<arg>5s</arg>
<doc>Pauses the test executed for the given time.</doc>
<status status="NOT RUN" starttime="20220908 18:07:50.821" endtime="20220908 18:07:50.821"/>
</kw>
<status status="FAIL" starttime="20220908 18:07:30.219" endtime="20220908 18:07:50.821"/>
</kw>
<kw name="login with credientials" library="robertApp">
<arg>appiumtesting7@gmail.com</arg>
<arg>appiumtesting7</arg>
<status status="NOT RUN" starttime="20220908 18:07:50.821" endtime="20220908 18:07:50.821"/>
</kw>
<kw name="Close app" library="KeywordsFile">
<status status="NOT RUN" starttime="20220908 18:07:50.821" endtime="20220908 18:07:50.821"/>
</kw>
<doc>This test case verifies whether we can successfully login with correct username and password</doc>
<status status="FAIL" starttime="20220908 18:07:30.219" endtime="20220908 18:07:50.821">WebDriverException: Message: Failed to execute command appium.mode start: appiumModeStart command failed. Reason: handset server: application is not installed
Stacktrace:
    at org.openqa.selenium.remote.server.PerfectoMobileSession$Factory.convertToRuntimeException (PerfectoMobileSession.java:166)
    at org.openqa.selenium.remote.server.PerfectoMobileSession$Factory.apply (PerfectoMobileSession.java:150)
    at org.openqa.selenium.remote.server.PerfectoMobileActiveSessionFactory.lambda$apply$3 (PerfectoMobileActiveSessionFactory.java:117)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:193)
    at java.util.stream.ReferencePipeline$11$1.accept (ReferencePipeline.java:373)
    at java.util.stream.ReferencePipeline$2$1.accept (ReferencePipeline.java:175)
    at java.util.Spliterators$ArraySpliterator.tryAdvance (Spliterators.java:958)
    at java.util.stream.ReferencePipeline.forEachWithCancel (ReferencePipeline.java:126)
    [ Message content over the limit has been removed. ]
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:471)
    at java.util.stream.FindOps$FindOp.evaluateSequential (FindOps.java:152)
    at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.findFirst (ReferencePipeline.java:464)
    at org.openqa.selenium.remote.server.NewSessionPipeline.createNewSession (NewSessionPipeline.java:72)
    at org.openqa.selenium.remote.server.commandhandler.BeginSession.execute (BeginSession.java:65)
    at org.openqa.selenium.remote.server.MobileWebDriverServlet.lambda$handle$2 (MobileWebDriverServlet.java:299)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:511)
    at java.util.concurrent.FutureTask.run (FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:617)
    at java.lang.Thread.run (Thread.java:748)</status>
</test>
<status status="FAIL" starttime="20220908 18:07:29.944" endtime="20220908 18:07:50.821"/>
</suite>
<statistics>
<total>
<stat pass="0" fail="1" skip="0">All Tests</stat>
</total>
<tag>
</tag>
<suite>
<stat pass="0" fail="1" skip="0" id="s1" name="perfecto Test">perfecto Test</stat>
</suite>
</statistics>
<errors>
<msg timestamp="20220908 18:07:30.191" level="ERROR">Error in file 'C:\Users\vuchander\POD5_Mobile_libs\Tests\perfecto_Test.robot' on line 3: Importing library 'PerfectoLibrary' failed: ModuleNotFoundError: No module named 'client'
Traceback (most recent call last):
  File "C:\Users\vuchander\AppData\Roaming\Python\Python310\site-packages\PerfectoLibrary\__init__.py", line 4, in &lt;module&gt;
    from PerfectoLibrary.keywords import *
  File "C:\Users\vuchander\AppData\Roaming\Python\Python310\site-packages\PerfectoLibrary\keywords\__init__.py", line 4, in &lt;module&gt;
    from ._devices import _DeviceKeywords
  File "C:\Users\vuchander\AppData\Roaming\Python\Python310\site-packages\PerfectoLibrary\keywords\_devices.py", line 1, in &lt;module&gt;
    import perfecto
  File "C:\Users\vuchander\AppData\Roaming\Python\Python310\site-packages\perfecto\__init__.py", line 1, in &lt;module&gt;
    from client import *
PYTHONPATH:
  C:\Program Files\Python310\Scripts\robot.exe
  C:\Program Files\Python310\python310.zip
  C:\Program Files\Python310\DLLs
  C:\Program Files\Python310\lib
  C:\Program Files\Python310
  C:\Users\vuchander\AppData\Roaming\Python\Python310\site-packages
  C:\Program Files\Python310\lib\site-packages</msg>
<msg timestamp="20220908 18:07:50.821" level="WARN">Keyword 'Capture Page Screenshot' could not be run on failure: No application is open</msg>
</errors>
</robot>
